localMatchApp/sources/homewindow.cpp

```diff
@@ -1,50 +1,152 @@
  * main system screen is destroyed
  */
-HomeWindow::~HomeWindow ( ) {
-  delete ui;
+
+HomeWindow::~HomeWindow(){}
+
+void HomeWindow::onButtonPreviousClicked()
+{
+    /*! A number is subtracted from the index to go to the previous event */
+    indexOfCarousel--;
+
+    /*! The function to show information in the carousel is called and the index value is sent */
+    showInformationCarousel(indexOfCarousel);
+
+    /*! The function to show buttons in the carousel is called and the index value is sent */
+    showButtonsCarousel(indexOfCarousel);
+}
+
+void HomeWindow::onButtonNextClicked()
+{
+    /*! A number is added to the index to go to the next event */
+    indexOfCarousel++;
+
+    /*! The function to show information in the carousel is called and the index value is sent */
+    showInformationCarousel(indexOfCarousel);
+
+    /*! The function to show buttons in the carousel is called and the index value is sent */
+    showButtonsCarousel(indexOfCarousel);
+}
+
+void HomeWindow::showInformationCarousel(int i_indexOfCarousel)
+{
+    /*!
+     *  It is assigned the values ​​brought from the function select event to the object
+     *  type event to be able to call the get function the values ​​return
+     */
+    event = functionsOfDatabase.selectEvent(i_indexOfCarousel);
+
+    /*! To the index of the carousel is added a number because it starts in zero */
+    eventStatus = QString::number(i_indexOfCarousel+1)+"/"+QString::number(event.getTotalEvents())+" of events";
+
+    /*! If there are no events a message is show to the user */
+    if(event.getTotalEvents() == NOEVENTS)
+    {
+        ui->labelNameEvent->setText("No Data");
+    }
+    /*! If there are at least one event the information of the user is showed */
+    else{
+        ui->labelNameEvent->setText (event.getNameEvent());
+        ui->labelDescription->setText (event.getDescriptionEvent());
+        ui->labelEventInitialDate->setText (event.getInitialDateEvent());
+        ui->labelEventFinalDate->setText (event.getFinalDateEvent());
+        ui->pictureEvent->setPixmap (event.getImageEvent());
+        ui->labelEventStatus->setText(eventStatus);
+    }
+}
+
+void HomeWindow::showButtonsCarousel(int i_indexOfCarousel)
+{
+    /*! If there are only one event the next and previous buttons are disabled */
+    if (event.getTotalEvents() == ONEEVENT)
+    {
+        ui->ButtonPrevious->setDisabled(true);
+        ui->ButtonNext->setDisabled(true);
+    }
+
+    /*! If the event is the first, the button previous is disabled */
+    if(indexOfCarousel==FIRSTEVENT)
+    {
+        ui->ButtonPrevious->setDisabled(true);
+    }else
+    {
+        ui->ButtonPrevious->setDisabled(false);
+    }
+
+    /*! If the event is the last, the button next is disabled */
+    /*! To the index of the carousel is substracted a number because the total of events
+    * is a number superior */
+    if(indexOfCarousel==(event.getTotalEvents()-1))
```

Comment:
Please merge the 3 IF statements into just one to enable/disable the buttons.
You just need to compare if there is just one event, or if we are in the first or last event.
So, with an IF, ELSE IF, ELSE statements you can do the same